app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-event-tracker.js

```diff
@@ -11,68 +11,78 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
   }
 
   Ga4EventTracker.prototype.init = function () {
-    this.module.addEventListener('click', this.trackClick.bind(this), true) // useCapture must be true
+    if (window.GOVUK.analyticsGA4.consent) {
```

Comment:
That's an excellent point. I'm not certain in which order our code executes, but this would fail in the following scenario.

- user consents to cookies on a previous page
- Modules code runs first, looks for cookie consent flag, finds false, does not execute analytics Module
- analytics code runs, sets consent flag to true based on previously set cookie

It does work if the analytics code runs first, but we can't guarantee that:

- user consents to cookies on a previous page
- analytics code runs, sets consent flag to true based on previously set cookie
- Modules code runs first, looks for cookie consent flag, finds true, executes

Fortunately this doesn't mean that tracking would happen without consent, but it does mean that when users consent we might miss out on tracking data.

This might be a bit of a thorny one to sort out right now, so I think the best option for the moment is to go back to the original model [already in use by some Modules](https://github.com/alphagov/govuk_publishing_components/blob/main/docs/javascript-modules.md#modules-and-cookie-consent), which is to use `window.GOVUK.getConsentCookie()` to check, rather than relying on a flag set by the analytics code.

This solution is less efficient (more than one bit of code would read the cookie) but for now I think the best way forward. I'll record this problem in our technical notes for later investigation.

Thanks for raising.